### PR TITLE
103590 loans

### PR DESCRIPTION
**Pull request by jarednimble**
Resolves:-
https://dfe-gov-uk.visualstudio.com/Academies-and-Free-Schools-SIP/_workitems/edit/103590/
https://dfe-gov-uk.visualstudio.com/Academies-and-Free-Schools-SIP/_workitems/edit/103858/

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Expected behaviour

You can progress the loan section under finances and the loans will be sent to the API

Added loan functionality to the school finances section

## Steps to test this PR

1. Go to the finance review page
2. Fill out the loans section
3. Confirm the data is saved in the database

## Prerequisites
n/a

### CODE REVIEW

File: Dfe.Academies.External.Web/ViewModels/LoanViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+﻿namespace Dfe.Academies.External.Web.ViewModels
+{
+	public class LoanViewModel
+	{
+		public int Id { get; set; }
+		public bool IsDraft { get; set; }
+
+		public string DisplayName { get; set; }
```
Comment (MarkRobinsonDfe) [inline Dfe.Academies.External.Web/ViewModels/LoanViewModel.cs:8]: do you want to do the = null!; at the end of the non-nullable string to remove these build warnings?

File: Dfe.Academies.External.Web/ViewModels/FinancesReviewSectionViewModel.cs [hunk 2]
```diff
@@ -26,7 +26,7 @@ public class FinancesReviewSectionViewModel : SchoolQuestionAndAnswerViewModel
 		public const string NFYCapitalCarryForwardExplained = ""; // ??
 
 		// loans
-		// TODO MR:- question string defs
+		public const string LExistingLoans = "Are there any existing loans?";
```
Comment (MarkRobinsonDfe) [inline Dfe.Academies.External.Web/ViewModels/FinancesReviewSectionViewModel.cs:29]: can we rename this const without the 'L' at the beginning? looks a bit odd

File: Dfe.Academies.External.Web/Pages/School/LoanDetails.cshtml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+				        Schedule of Repayment
+				        </label>
+			        <textarea asp-for="RepaymentSchedule" id="RepaymentSchedule" class="govuk-textarea" cols="40" rows="5" maxlength="400"></textarea>
+		        </div>                
+		        
+	        </fieldset>
+		</div>
+		<br />
+        
+		<input type="submit" value="Save and return to overview" class="govuk-button govuk-!-margin-top-2 govuk-!-margin-bottom-8"/>
```
Comment (MarkRobinsonDfe) [inline Dfe.Academies.External.Web/Pages/School/LoanDetails.cshtml:68]: before dan tests this, is this button text correct? Should it be 'continue' or 'save and continue'?

File: Dfe.Academies.External.Web/Pages/School/LoanDetails.cshtml.cs [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+	public class LoanDetails : BasePageEditModel
+	{
+		[BindProperty]
+		public int  Id { get; set; }
+		[BindProperty]
+		public int ApplicationId { get; set; }
+
+		[BindProperty]
+		public int Urn { get; set; }
+		[BindProperty]
```
Comment (MarkRobinsonDfe) [inline Dfe.Academies.External.Web/Pages/School/LoanDetails.cshtml.cs:19]: the validation text isn't as per the ticket if you don't fill these in. just spotting before dan does

**Review by MarkRobinsonDfe — CHANGES_REQUESTED** [review]
few comments to fix up.

Also, I seem to be able to break the caching if I go in and out of the loans page:-

image

File: Dfe.Academies.External.Web/Pages/School/LoanDetails.cshtml (hunk 3 above)
Comment (jarednimble) [reply Dfe.Academies.External.Web/Pages/School/LoanDetails.cshtml:68]: The ticket shows 'save and continue' on the picture so I used that. Seems like the AC and the example are different

**Review by MarkRobinsonDfe — APPROVED** [review]
all good now. go ahead & merge

**Pull request merged**